client/components/data/query-memberships/index.js

```diff
@@ -15,39 +15,24 @@ import { requestProducts } from 'state/memberships/product-list/actions';
 
 class QueryMemberships extends Component {
 	static propTypes = {
-		productId: PropTypes.number,
 		siteId: PropTypes.number.isRequired,
-		requestProduct: PropTypes.func,
 		requestProducts: PropTypes.func,
 	};
 
 	static defaultProps = {
 		productId: null,
 	};
 
-	UNSAFE_componentWillMount() {
-		this.request( this.props );
-	}
-
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( nextProps.siteId !== this.props.siteId || nextProps.productId !== this.props.productId ) {
-			this.request( nextProps );
-		}
-	}
-
-	request( props ) {
-		const { siteId, productId } = props;
-
-		if ( ! siteId ) {
+	componentDidMount() {
+		if ( this.props.requesting ) {
```

Comment:
not needed in this PR since we're targeting the `UNSAFE_` methods only but just a note that in the end this `requesting` boolean should disappear